src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -21,10 +22,13 @@ async function assume_role(req) {
         }
         throw new StsError(StsError.InternalFailure);
     }
+
+    const { expiration, seconds_until_expiration } = _get_expiration_data(req);
```

Comment:
I think this parsing should be done before calling sts_sdk.get_assumed_role.
I would call like this:
```js
const duration_ms = parse_sts_duration(req.body.duration_seconds);
const expiration_date = new Date(Date.now() + duration_ms);
```

Reply:
Also I see that our http_utils parses CONTENT_TYPE_APP_FORM_URLENCODED and converts all keys with _.snakeCase. I think this is a bad idea which can allow unintended keys to be used. Can we get rid of it? This is only used by sts and iam so not a lot of cases to fix.

Reply:
@guymguym IAM service was written based on STS, so if it's in STS it's in IAM...
I didn't see any problem with this function (I printed the keys during debugging), for example:

> Aug-27 7:41:24.015 [nsfs/3080]   [LOG] CONSOLE:: SDSD req.body Action=CreateUser&Version=2010-05-08&Path=%2Fabc%2Fdef%2F&UserName=Benny
> Aug-27 7:41:24.016 [nsfs/3080]   [LOG] CONSOLE:: SDSD res [Object: null prototype] { Action: ‘CreateUser’, Version: ‘2010-05-08’, Path: ‘/abc/def/’, UserName: ‘Benny’ }
> Aug-27 7:41:24.017 [nsfs/3080]   [LOG] CONSOLE:: SDSD renamed { action: ‘CreateUser’, version: ‘2010-05-08’, path: ‘/abc/def/’, user_name: ‘Benny’ }

If you decide to change it, it should be in a different PR.